### ISSUE TITLE
refactor: add ghost-ring zeroing utility functions

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#158]
## Context

The codebase has ~135 instances of the pattern:
```python
out = jnp.zeros_like(x)
out = out.at[interior_slice].set(values)
```
to enforce the ghost-ring convention (interior values surrounded by zeros). There's also a z-ghost zeroing pattern (`.at[0].set(0.0).at[-1].set(0.0)`) used 8 times after `jax.vmap` in 3D operators.

A small utility module would DRY this up, make the convention explicit, and reduce the chance of ghost-ring bugs.

---

### `interior(values, like, ghost=1)` — embed interior values in a zero-padded array

```python
def interior(values, like, ghost=1):
    """Create an array shaped like ``like``, zero everywhere except the interior."""
    slices = tuple(slice(ghost, -ghost) for _ in range(like.ndim))
    out = jnp.zeros(like.shape, dtype=jnp.result_type(like, values))
    return out.at[slices].set(values)
```

Covers:
- `[1:-1]` (1D, ghost=1) — 15 instances
- `[1:-1, 1:-1]` (2D, ghost=1) — 76 instances
- `[1:-1, 1:-1, 1:-1]` (3D, ghost=1) — 33 instances
- `[2:-2, 2:-2]` / `[2:-2]` (ghost=2) — 4 instances

### `zero_z_ghosts(arr)` — zero first and last z-slices

```python
def zero_z_ghosts(arr):
    """Zero the first and last slices along axis 0."""
    return arr.at[0].set(0.0).at[-1].set(0.0)
```

Covers: 8 instances across 4 files (all 3D vmap operators).

### What stays as explicit slices (NOT converted)

- **Asymmetric flux** `[1:-1, 1:-2]` / `[1:-2, 1:-1]` in diffusion.py (4 instances) — asymmetry is load-bearing
- **3D mixed** `[1:-1, 2:-2, 2:-2]` in advection.py (1 instance) — mixed ghost widths per axis
- **Layer passthrough** `[:, 1:-1, 1:-1]` in diagnostics.py (2 instances) — leading `:` has distinct semantics

---

## File location

`finitevolx/_src/operators/_ghost.py` — internal helper, not added to public API.

---

### Commit 1: Add `_ghost.py` + unit tests
- Create `finitevolx/_src/operators/_ghost.py` (~30 lines)
- Create `tests/test_ghost.py` — test 1D/2D/3D shapes, ghost=1 and ghost=2, dtype inference, JIT compatibility, `zero_z_ghosts` preserves interior

### Commit 2: Convert z-ghost pattern (8 instances, 4 files)
- `finitevolx/_src/operators/vorticity.py` (1)
- `finitevolx/_src/operators/coriolis.py` (2)
- `finitevolx/_src/diffusion/diffusion.py` (3)
- `finitevolx/_src/diffusion/momentum.py` (2)

Each: `.at[0].set(0.0).at[-1].set(0.0)` → `zero_z_ghosts(...)`

### Commit 3: Convert 2D `[1:-1, 1:-1]` interior pattern
Target files: `vorticity.py`, `coriolis.py`, `momentum.py`, `diffusion.py`, `diagnostics.py`, `geographic.py`, `difference.py`, `interpolation.py`, and 2D methods in `reconstruction.py`, `flux.py`

Each: `out = jnp.zeros_like(x); out = out.at[1:-1, 1:-1].set(expr)` → `out = interior(expr, x)`

### Commit 4: Convert 1D and 3D standard interior patterns
- 1D `[1:-1]` in reconstruction.py, difference.py, interpolation.py
- 3D `[1:-1, 1:-1, 1:-1]` in difference.py, interpolation.py, reconstruction.py
- Strict interior `[2:-2, ...]` (ghost=2) in advection.py, momentum.py

### Commit 5: Lint/format cleanup

---

## Verification

```bash
uv run pytest tests -v            # All tests pass
uv run ruff check .               # Lint clean
uv run ruff format --check .      # Format clean
uv run ty check finitevolx        # Typecheck clean
```
